src/easylogging++.cc

```diff
@@ -1717,17 +1717,23 @@ void TypedConfigurations::insertFile(Level level, const std::string& fullFilenam
   auto create = [&](Level level) {
     base::LogStreamsReferenceMap::iterator filestreamIter = m_logStreamsReference->find(resolvedFilename);
     base::type::fstream_t* fs = nullptr;
-    if (filestreamIter == m_logStreamsReference->end()) {
+    bool needNewStream = filestreamIter == m_logStreamsReference->end();
+    if (!needNewStream && filestreamIter->second.lock() == nullptr) {
+      // Filestream is expired and exists in map
+      needNewStream = true;
+      m_logStreamsReference->erase(filestreamIter);
```

Comment:
There may still be some remaining elements in map that can't be removed

Reply:
I don't get `that can't be removed`.
This erase only one expired file stream.
If it is not expired, why remove?